src/resolve/local.ts

```diff
@@ -17,61 +13,35 @@ export default async function resolveLocal (spec: PackageSpec, opts: ResolveOpti
 
   if (dependencyPath.slice(-4) === '.tgz' || dependencyPath.slice(-7) === '.tar.gz') {
     const name = getTarballName(dependencyPath)
-    return {
+    const resolution: Resolution = {
+      type: 'tarball',
       id: createLocalPkgId(name, dependencyPath),
-      root: path.dirname(dependencyPath),
-      fetch: createLocalTarballFetcher({
-        tarball: dependencyPath
-      })
+      tarball: `file:${dependencyPath}`,
     }
+    return {resolution}
   }
 
   if (opts.linkLocal) {
```

Comment:
Seems like this if statement is totally redundant after moving out the fetch. The logic is the same. I think

1. this if statement can be removed
2. the link boolean can be removed from the DirectoryResolution type
3. the linkLocal option can be used in the fetch function to decide how to connect the dependency to the project